src/Compilers/CSharp/Test/Emit3/Semantics/ExtensionTests2.cs

```diff
@@ -32647,6 +32649,151 @@ .method family hidebysig static void M () cil managed
         Assert.Null(method.AssociatedExtensionImplementation);
     }
 
+    [Fact]
+    public void AssociatedExtensionImplementation_06()
+    {
+        // not a definition, generic extension and method
+        var src = """
+42.M(43L, "");
+
+public static class E
+{
+    extension<T>(T t)
+    {
+        public void M<U, V>(U u, V v) { }
+    }
+}
+""";
+        var comp = CreateCompilation(src);
+        comp.VerifyEmitDiagnostics();
+
+        var tree = comp.SyntaxTrees.First();
+        var model = comp.GetSemanticModel(tree);
+        var memberAccess = GetSyntax<MemberAccessExpressionSyntax>(tree, "42.M");
+        var method = (IMethodSymbol)model.GetSymbolInfo(memberAccess).Symbol;
+        AssertEx.Equal("void E.M<System.Int32, System.Int64, System.String>(this System.Int32 t, System.Int64 u, System.String v)",
+            method.AssociatedExtensionImplementation.ToTestDisplayString());
+    }
+
+    [Fact]
+    public void AssociatedExtensionImplementation_07()
+    {
+        // generic definition
+        var src = """
+public static class E
+{
+    extension<T>(T t)
+    {
+        public void M<U, V>(U u, V v)
+        {
+            t.M(u, v);
+        }
+    }
+}
+""";
+        var comp = CreateCompilation(src);
+        comp.VerifyEmitDiagnostics();
+
+        var tree = comp.SyntaxTrees.First();
+        var model = comp.GetSemanticModel(tree);
+        var memberAccess = GetSyntax<MemberAccessExpressionSyntax>(tree, "t.M");
+        var method = (IMethodSymbol)model.GetSymbolInfo(memberAccess).Symbol;
+        Assert.True(method.IsDefinition);
+        var associated = method.AssociatedExtensionImplementation;
```

Comment:
If we have a definition, why couldn't its `AssociatedExtensionImplementation` return definition too? The API review says it should be consistent with `AssociatedSymbol` and isn't that what it would do?

Reply:
> If the behavior of the API were to associate `E.M<T'>(T' t)` (definition) with `E.extension<T>(T t).M()` (definition), that would be irregular.

I see, makes sense, thanks. (I was already in the API review email thread, but this reasoning wasn't clear from the email to me.)

> Considering more than one type parameter, what would be expected if you had a partial substitution like `E2.M<T1', int>()`. Would it associate with `E2.extension<T1, int>().M()` or `E2.extension<T1', int>().M()`?

Do we have a test for that?